provider/ec2/provider.go

```diff
@@ -33,8 +34,17 @@ func (p environProvider) Open(args environs.OpenParams) (environs.Environ, error
 	e.cloud = args.Cloud
 	e.name = args.Config.Name()
 
+	// The endpoints in public-clouds.yaml from 2.0-rc2
+	// and before were wrong, so we use whatever is defined
+	// in goamz/aws if available.
+	if isBrokenCloud(e.cloud) {
```

Comment:
Just to check - is this the only place we need to do this? It covers add-model as well as bootstrap? Can you confirm the add-model case has been tested?